### PR TITLE
Support Swagger

### PR DESCRIPTION
**Pull request by moarychan**
## Purpose
Support view and test using the Swagger UI page, integrated the authorization code flow with Azure AD to acquire access token.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

Follow this [quickstart doc](https://review.learn.microsoft.com/en-us/azure/spring-apps/quickstart-deploy-restful-api-app?branch=pr-en-us-244572&tabs=Azure-portal#4-deploy-the-app-to-azure-spring-apps) to test.

## What to Check
Verify that the following are valid
* ...

## Other Information

N/A

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -59,6 +59,11 @@
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.springdoc</groupId>
+      <artifactId>springdoc-openapi-ui</artifactId>
+      <version>1.7.0</version>
```
Comment (saragluna) [inline pom.xml]: use property for this version

File: src/main/java/com/microsoft/azure/simpletodo/configuration/WebConfiguration.java [hunk 2]
```diff
@@ -21,4 +44,36 @@ public void addCorsMappings(CorsRegistry registry) {
...
         };
     }
+
+    @Bean
+    public WebSecurityCustomizer webSecurityCustomizer() {
+        return (web) -> web.ignoring().antMatchers("/", "/swagger-ui.html", "/swagger-ui/**", "/v3/api-docs", "/v3/api-docs/**");
+    }
+
+    @Bean
+    public OpenAPI customOpenAPI() {
```
Comment (saragluna) [inline src/main/java/com/microsoft/azure/simpletodo/configuration/WebConfiguration.java]: Can we move all swagger configuration to a separate configuration class?

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -15,6 +18,19 @@ spring:
...
       active-directory:
         enabled: true
+        profile:
+          tenant-id: <your-Azure-AD-tenant-ID>
         credential:
           client-id: <your-application-ID-of-ToDo>
         app-id-uri: <your-application-ID-URI-of-ToDo>
+
+spring-apps:
+  public-url: ${SPRING_APPS_PUBLIC_URL:http://localhost:8080}
```
Comment (saragluna) [inline src/main/resources/application.yml]: who will set this env SPRING_APPS_PUBLIC_URL?

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -1,3 +1,6 @@
+server:
+  tomcat:
+    use-relative-redirects: true
```
Comment (saragluna) [inline src/main/resources/application.yml]: What's this for?

File: src/main/resources/application.yml (hunk 3 above)
Comment (saragluna) [reply src/main/resources/application.yml]: Can we move all properties related to swagger together, and add some comments for others to understand it's for swagger?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml (hunk 3 above)
Comment (moarychan) [reply src/main/resources/application.yml]: > who will set this env SPRING_APPS_PUBLIC_URL?

After the app deployment, then can get the exposed URL, it's required to set the API server URL, or else the HTTP request will route to the HTTP schema, not the HTTPS.

File: src/main/resources/application.yml (hunk 4 above)
Comment (moarychan) [reply src/main/resources/application.yml]: > What's this for?

It's used to redirect with HTTPS, when going to `https://simple-todo-api.xxx.azurecontainerapps.io`， then  redirect to `http://simple-todo-api.xxx.azurecontainerapps.io/swagger-ui/index.html`.

File: src/main/resources/application.yml (hunk 3 above)
Comment (saragluna) [reply src/main/resources/application.yml]: So if I deploy the app to ASA, I need to set the env, and then restart the app?

Comment (moarychan) [reply src/main/resources/application.yml]: Yes, the oauth2 redirect url should be updated to use the app exposed URL as prefix.

**Review by saragluna — APPROVED** [review]

**Pull request merged**